commit 7725e7bfe6f2ce8146b6552b44e0d226be7638e7
Author: Pauli <pauli@openssl.org>
Date:   Fri Nov 11 09:40:19 2022 +1100

    x509: fix double locking problem
    
    This reverts commit 9aa4be691f5c73eb3c68606d824c104550c053f7 and removed the
    redundant flag setting.
    
    Fixes #19643
    
    Fixes LOW CVE-2022-3996
    
    Reviewed-by: Dmitry Belyavskiy <beldmit@gmail.com>
    Reviewed-by: Tomas Mraz <tomas@openssl.org>
    (Merged from https://github.com/openssl/openssl/pull/19652)
    
    (cherry picked from commit 4d0340a6d2f327700a059f0b8f954d6160f8eef5)

diff --git a/crypto/x509/pcy_map.c b/crypto/x509/pcy_map.c
index 05406c6493..60dfd1e320 100644
--- a/crypto/x509/pcy_map.c
+++ b/crypto/x509/pcy_map.c
@@ -73,10 +73,6 @@ int ossl_policy_cache_set_mapping(X509 *x, POLICY_MAPPINGS *maps)
 
     ret = 1;
  bad_mapping:
-    if (ret == -1 && CRYPTO_THREAD_write_lock(x->lock)) {
-        x->ex_flags |= EXFLAG_INVALID_POLICY;
-        CRYPTO_THREAD_unlock(x->lock);
-    }
     sk_POLICY_MAPPING_pop_free(maps, POLICY_MAPPING_free);
     return ret;
 
